devlib/utils/ssh.py

```diff
@@ -170,51 +305,473 @@ def name(self):
+
+        if use_sudo:
+            if self._sudo_needs_password and not self.password:
+                raise TargetStableError('Attempt to use sudo but no password was specified')
+
+            command = self.sudo_cmd.format(quote(command))
+
+            log_debug(command)
+            streams = executor(command, timeout=timeout)
+            if self._sudo_needs_password:
+                stdin = streams[0]
+                stdin.write(self.password + '\n')
+                stdin.flush()
+        else:
+            log_debug(command)
+            streams = executor(command, timeout=timeout)
+
+        return streams
+
+    def _execute(self, command, timeout=None, as_root=False, strip_colors=True, log=True):
+        # Merge stderr into stdout since we are going without a TTY
+        command = '({}) 2>&1'.format(command)
+
+        stdin, stdout, stderr = self._execute_command(
+            command,
+            as_root=as_root,
+            log=log,
+            timeout=timeout,
+            executor=self.client.exec_command,
+        )
+        stdin.close()
+
+        # Empty the stdout buffer of the command, allowing it to carry on to
+        # completion
+        def callback(output_chunks, name, chunk):
+            output_chunks.append(chunk)
+            return output_chunks
+
+        select_timeout = 1
+        output_chunks, exit_code = _read_paramiko_streams(stdout, stderr, select_timeout, callback, [])
+        # Join in one go to avoid O(N^2) concatenation
+        output = b''.join(output_chunks)
+
+        if sys.version_info[0] == 3:
+            output = output.decode(sys.stdout.encoding or 'utf-8', 'replace')
+        if strip_colors:
+            output = strip_bash_colors(output)
+
+        return (exit_code, output)
+
+
+class TelnetConnection(SshConnectionBase):
+
+    default_password_prompt = '[sudo] password'
+    max_cancel_attempts = 5
+
+    # pylint: disable=unused-argument,super-init-not-called
+    def __init__(self,
+                 host,
+                 username,
+                 password=None,
+                 port=None,
+                 timeout=None,
+                 password_prompt=None,
+                 original_prompt=None,
+                 sudo_cmd="sudo -- sh -c {}",
+                 strict_host_check=True,
+                 platform=None):
+
+        super().__init__(
+            host=host,
+            username=username,
+            password=password,
+            keyfile=None,
+            port=port,
+            platform=platform,
+            sudo_cmd=sudo_cmd,
+            strict_host_check=strict_host_check,
+        )
+
+        if self.strict_host_check:
+            options = {
+                'StrictHostKeyChecking': 'yes',
+            }
+        else:
+            options = {
+                'StrictHostKeyChecking': 'no',
+                'UserKnownHostsFile': '/dev/null',
+            }
+        self.options = options
```

Comment:
I'm not sure, the goal was to try to reproduce the current behavior since I don't have any telnet target at hand. If you think it's useless I can just remove it

Reply:
made me realize that I introduced a copy of `ssh_get_shell`, so I renamed it into `telnet_get_shell` and specialized it for telnet by removing irrelevant parameters.